tests/cli.rs

```diff
@@ -0,0 +1,280 @@
+#![cfg(any(target_os = "windows", target_os = "macos", target_os = "linux"))]
+use std::env;
+use std::fs::File;
+use std::io::{BufRead, BufReader, Read};
+use std::path::{Path, PathBuf};
+use std::process::{Child, Command, Stdio};
+
+use anyhow::{Context, Result};
+use rand::{RngCore, SeedableRng};
+use tempfile::tempdir;
+
+const ADDR: &str = "127.0.0.1:0";
+
+fn make_rand_file(size: usize, path: &Path) -> Result<()> {
+    let mut content = vec![0u8; size];
+    rand::rngs::StdRng::seed_from_u64(1).fill_bytes(&mut content);
+    std::fs::write(path, content)?;
+    Ok(())
+}
+
+#[test]
+fn cli_provide_one_file() -> Result<()> {
+    let dir = tempdir()?;
```

Comment:
So I'm still curious if this was a real bug in testdir that needs fixing?

Reply:
hi! just re-ran the tests to be sure: https://github.com/n0-computer/iroh/actions/runs/4264055450/jobs/7421605558

Yes, this is a bug:

"thread 'cli_provide_folder' panicked at 'Failed to create testdir: Failed to remove obsolete testdir-current symlink"

The windows machines aren't able to remove the `testdir-current` symlink in these cases, even though it works just fine when `testdir` is used as part of the code testing the library.

Reply:
thanks!